profiles/car.lua

```diff
@@ -606,9 +701,45 @@ function way_function (way, result)
       result.backward_speed = result.backward_speed * scaling_factor
     end
   end
+end
 
+-- determine if this way can be used as a start/end point for routing
+function handle_startpoint(way,result)
   -- only allow this road as start point if it not a ferry
-  result.is_startpoint = result.forward_mode == mode.driving or result.backward_mode == mode.driving
+  result.is_startpoint = result.forward_mode == mode.driving or 
+                              result.backward_mode == mode.driving
+end
+
+-- set the road classification based on guidance globals configuration
+function handle_classification(way,result)
+  set_classification(way:get_value_by_key("highway"),result,way)
+end
+
+-- main entry point for processsing a way
+function way_function(way, result)
+  -- table of temporary values, e.g. computed access status
+  local data = {}
+  
+  -- perform each procesing step sequentially.
+  -- most steps can abort processing, meaning the way
+  -- is not routable  
+  
+  if handle_initial_check(way,result,data) == false then return end
+  if handle_default_mode(way,result) == false then return end
+  if handle_blocking(way,result) == false then return end
+  if handle_access(way,result,data) == false then return end
+  if handle_hov(way,result) == false then return end
+  if handle_ferries(way,result) == false then return end
+  if handle_movables(way,result) == false then return end
+  if handle_service(way,result) == false then return end
+  if handle_oneway(way,result) == false then return end
+  if handle_speed(way,result,data) == false then return end
+  if handle_turn_lanes(way,result) == false then return end
+  if handle_junctions(way,result) == false then return end
+  if handle_startpoint(way,result) == false then return end
+  if handle_restricted(way,result,data) == false then return end
+  if handle_classification(way,result) == false then return end
+  if handle_names(way,result) == false then return end
```

Comment:
The function itself should be more descriptive. Here I would prefer having a single sentence giving you the general idea (as long as the function doesn't tell you).
E.g. if a function is called `extract_names`, I don't need a comment saying "Extracts road names from the way and assigns them", but `handle_initial_check` could benefit a lot from "Check if the way is admissible for navigation" or similar.